### DISCUSSION TITLE
Feature request:  a 'warning' mode built into the Stepper

### DISCUSSION THREAD

**Discussion by Snacksnacks** [vmware/clarity#5938] [Ideas]
This is a feature request to have a 'warning' mode built into the Stepper component. Our use case is that we want to convey there being a warning alert attached to one of the stepper steps, but cannot use the error mode stepper because it's technically not an error. 

More context:
In the Workload Management enablement flow, users can select the DHCP network mode during the management network step. This network mode will call for their DHCP server to provide values for some of the management network steps. The caveat is that their DHCP server may fail to acquire those values and leaving those fields blank may result in the enablement failing, which they won't know until they complete the entire enablement flow(see mockup below for the 'Review and Confirm' step) because we currently don't have a validation check for the DHCP server. 

That's why we want to show a warning alert in the management network step that tells the user they need to make sure their DHCP server supports Client Identifiers as per DHCP protocol/RFC and that their DHCP provides a compatible DNS server(s), NTP, and DNS search domain, otherwise risk enablement failure. We would want this warning to show when the stepper is collapsed and completed.

Mockup here:
![Screen_Shot_2021-05-11_at_2 26 28_PM](https://user-images.githubusercontent.com/44950405/117887051-f7df1c80-b264-11eb-8ddc-e64aeab3ebf9.png)

Thank you!

**Comment by colinreedmiller** [comment]
This sounds like a sensible addition. Would the step indicate success when the user completes it and then update to warning when the enablement does not complete successfully?

I notice in your example that the stepper shown indicates step-status with color alone (the left bar is green but there is no other indication that it is complete). The component includes icons for status on stepper steps - the number changes to the icon when the step is completed - and this would be necessary for a11y in the case you describe and may need to be remedied in your design and/or build.
Demo: [https://v4.clarity.design/documentation/stepper](https://v4.clarity.design/documentation/stepper)

**Reply by Snacksnacks** [reply]
For this release, we would have to show the stepper in the warning mode even after the user completes this step because we can't verify if those provided fields are valid until they complete the entire enablement wizard (i.e. clicking 'Finish' in the Review and Confirm step is the only time we can trigger a validation check currently). So because we can't check if their inputs are valid or not, we would have to show the warning alert regardless of what their inputs are.

re: the icons, thanks for pointing that out! Our builds have this implemented already but we will certainly update the designs accordingly.
